ycmd: unstable-2023-11-06 -> unstable-2025-06-16

S0AndS0 opened the pull request:
It was discovered, while struggling with the YouCompleteMe plugin/package [Issue](https://github.com/NixOS/nixpkgs/issues/429485), that sometime within the past two years the Python team made good on threats to remove `cgi` from the standard library.

This set of changes adds `legacy-cgi` to the `buildInputs` because the `ycmd` package needs `cgi` and, because I was already doing stuff, I updated the `rev`, `sha256`, and `version` attributes too ;-)

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [ ] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [ ] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

S0AndS0 committed:
maintainers: add S0AndS0

acid-bong commented on pkgs/by-name/yc/ycmd/package.nix:
```diff
@@ -28,8 +28,8 @@ stdenv.mkDerivation {
-    rev = "0607eed2bc211f88f82657b7781f4fe66579855b";
-    hash = "sha256-SzEcMQ4lX7NL2/g9tuhA6CaZ8pX/DGs7Fla/gr+RcOU=";
+    rev = "9160b4eee67ea61c8501bad36d061bcec5340021";
+    sha256 = "MSzYX1vXuhd4TNxUfHWaRu7O0r89az1XjZBIZ6B3gBk=";
```
Nope, `hash` is preferred nowadays

acid-bong reviewed (changes requested):
Moin and welcome to Nixpkgs. This PR of yours must be divided in 3 commits (the following aren't title examples btw):
- version bump + dependencies
- new maintainer entry 
- add yourself to the package maintainers

Any other edit should be amended to/squashed with its respective commit. You can force-push the resulting commits into the branch without closing the PR.

S0AndS0 committed:
ycmd: add maintainer S0AndS0

S0AndS0 commented:
> Moin and welcome to Nixpkgs. This PR of yours must be divided in 3 commits (the following aren't title examples btw):
> 
>     * version bump + dependencies
> 
>     * new maintainer entry
> 
>     * add yourself to the package maintainers
> 
> 
> Any other edit should be amended to/squashed with its respective commit. You can force-push the resulting commits into the branch without closing the PR.

Thanks for the rapid response!

I've got the commit history adjusted as instructed, titles be based on examples found in commit history, and have corrected the `sha256`/`hash` attribute...  and tested things still "work on my machine" too ;-)

If there be any other mistakes I've made, then please do call me out on 'em!

SuperSandro2000 commented on pkgs/by-name/yc/ycmd/package.nix:
```diff
@@ -21,15 +21,15 @@
   pname = "ycmd";
-  version = "unstable-2023-11-06";
+  version = "unstable-2025-06-16";
   disabled = !python3.isPy3k;
```
```suggestion
```
python2 is super duper EOL at this point

S0AndS0 committed:
ycmd: unstable-2023-11-06 -> unstable-2025-06-16

Update `version`, `rev`, and `sha256` because why not update if also
fixing dependency renaming stuff.

Add `legacy-cgi` due to breaking changes in Python updates;

- https://pypi.org/project/legacy-cgi/
- https://peps.python.org/pep-0594/#deprecated-modules
- https://peps.python.org/pep-0594/#cgi

TLDR: `cgi` was deprecated in version `3.11` and aimed to be removed in
`3.12` of Python due to;

> “[...] designed poorly and are now near-impossible to fix (`cgi`)
> [...]”

Remove `disable` attribute at request of @SuperSandro2000 because
Python2 is EOL

Add symbolic link to address Issue #429485 with YouCompleteMe Vim
plugin, however, beware there are _opportunities_ for future bug reports
to be opened because of `ycmd` and YouCompleteMe version de-sync.

Add `longDescription` with notes about YouCompleteMe errors when
run-time and compile-time Python interpreters do not match.

S0AndS0 commented:
> Nah, you're good now. Just rebase onto a fresher master commit and resolve conflicts

Woot!  I think I got it sorted then, though probably best to not trust my thinking just yet,,, rebasing with conflicts ain't always a clean process x-)

acid-bong commented on pkgs/by-name/yc/ycmd/package.nix:
```diff
@@ -21,15 +21,14 @@
   pname = "ycmd";
-  version = "0-unstable-2023-11-06";
-  disabled = !python3.isPy3k;
+  version = "unstable-2025-06-16";
```
ah yes, you shouldn't remove the 0, the current versioning convention says it should be there: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#versioning

acid-bong reviewed (approved):
Approved automatically following the successful run of `nixpkgs-review`.

---

these messages were autogenerated by Actions, which I ran before writing the latest comment. the said comment stays in power btw

SuperSandro2000 commented on pkgs/by-name/yc/ycmd/package.nix:
```diff
@@ -84,6 +84,16 @@ stdenv.mkDerivation {
     ln -s $out/lib/ycmd/ycmd/__main__.py $out/bin/ycmd
 
+    ## Work-around CMake/Nix naming of `.so` output
+    ln -s $out/lib/ycmd/ycm_core.cpython-[[:digit:]-][^[:space:]]*-gnu.so $out/lib/ycmd/ycm_core.so
```
Since we do not want to break darwin, we should use ``stdenv.hostPlatform.extensions.sharedLibrary`` or make it conditional on linux 😅

S0AndS0 commented:
Quick check-in with y'all, is there anything I need to do on my end to ensure things are easy to review?
